code/modules/mob/living/living.dm

```diff
@@ -452,6 +452,48 @@ default behaviour is:
 
 	return
 
+// This function is not to be invoked when the mob is killed. It runs when the mob is leaving the game or similar means.
+/mob/living/despawn()
+	//Update any existing objectives involving this mob.
+	for(var/datum/objective/O in all_objectives)
+		// We don't want revs to get objectives that aren't for heads of staff. Letting
+		// them win or lose based on cryo is silly so we remove the objective.
+		if(O.target == src.mind)
+			if(O.owner && O.owner.current)
+				to_chat(O.owner.current, SPAN_WARNING("You get the feeling your target is no longer within your reach..."))
+			qdel(O)
+
+	//Same for contract-based objectives.
+	for(var/datum/antag_contract/contract in GLOB.all_antag_contracts)
+		contract.on_mob_despawned(src.mind)
+
+	//Handle job slot/tater cleanup.
+	var/job = src.mind.assigned_role
```

Comment:
Might want to check if there's a mind first

Reply:
With if(src.mind) check ?

Reply:
yes

Reply:
Done